mmv1/third_party/terraform/tests/data_source_dns_managed_zone_test.go.erb

```diff
@@ -2,62 +2,91 @@
-		PreCheck:     func() { testAccPreCheck(t) },
-		ProtoV5ProviderFactories: map[string]func() (tfprotov5.ProviderServer, error){
-			"google": func() (tfprotov5.ProviderServer, error) {
-				provider, err := MuxedProviders(t.Name())
-				return provider(), err
-			},
-		},
+		PreCheck: func() { testAccPreCheck(t) },
 		// CheckDestroy: testAccCheckDNSManagedZoneDestroyProducer(t),
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDataSourceDnsManagedZone_basic(randString(t, 10)),
-				Check: checkDataSourceStateMatchesResourceStateWithIgnores(
-					"data.google_dns_managed_zone.qa",
-					"google_dns_managed_zone.foo",
-					map[string]struct{}{
-						"dnssec_config.#":                       {},
-						"private_visibility_config.#":           {},
-						"peering_config.#":                      {},
-						"forwarding_config.#":                   {},
-						"force_destroy":                         {},
-						"labels.#": 				                     {},
-						"creation_time": 			                   {},
-						"cloud_logging_config.#":                {},
-						"cloud_logging_config.0.%":              {},
-						"cloud_logging_config.0.enable_logging": {},
+				ExternalProviders: providerVersion450(),
+				Config:            testAccDataSourceDnsManagedZone_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceStateWithIgnores(
+						"data.google_dns_managed_zone.qa",
+						"google_dns_managed_zone.foo",
+						map[string]struct{}{
+							"dnssec_config.#":                       {},
+							"private_visibility_config.#":           {},
+							"peering_config.#":                      {},
+							"forwarding_config.#":                   {},
+							"force_destroy":                         {},
+							"labels.#":                              {},
+							"creation_time":                         {},
+							"cloud_logging_config.#":                {},
+							"cloud_logging_config.0.%":              {},
+							"cloud_logging_config.0.enable_logging": {},
+<% unless version == "ga" -%>
+							"reverse_lookup":                        {},
+<% end -%>
+						},
+					),
+					testExtractResourceAttr("data.google_dns_managed_zone.qa", "dns_name", &dns_name1),
+				),
+			},
+			{
+				ProtoV5ProviderFactories: protoV5ProviderFactories(t),
+				Config:                   testAccDataSourceDnsManagedZone_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceStateWithIgnores(
+						"data.google_dns_managed_zone.qa",
+						"google_dns_managed_zone.foo",
+						map[string]struct{}{
+							"dnssec_config.#":                       {},
+							"private_visibility_config.#":           {},
+							"peering_config.#":                      {},
+							"forwarding_config.#":                   {},
+							"force_destroy":                         {},
+							"labels.%":                              {}, // Note - requires change from labels.# to labels.% for muxed provider
```

Comment:
This necessary change to make the test pass for the muxed provider is a bit confusing.

@megan07 do you know what `%` and `#` signify in test state?